pkg/cloud/services/compute/instance.go

```diff
@@ -235,7 +235,6 @@ func (s *Service) createInstanceImpl(eventObject runtime.Object, openStackCluste
 		AvailabilityZone: instanceSpec.FailureDomain,
 		Networks:         portList,
 		UserData:         []byte(instanceSpec.UserData),
-		SecurityGroups:   securityGroups,
```

Comment:
I assume you have follow up PR to separate the sec group into port level?

Reply:
@jichenjc 

All the security groups are actually already propagated to the ports! Prior to this patch they are just duplicated on the instance as well.

Reply:
ah.. thanks for the info
yes, the port definition has recently been updated to add the sec group info already

but do we want to update `instanceSpec.SecurityGroups` since we don't honor 
it at all?

Reply:
@jichenjc 

`instanceSpec` is actually used to propagate the "instance default security groups" to the ports, as in the secgroups that should be used if the port does not specify its own (or disable port security).

See the following chain of usage:

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/6859acb599b3c358e682da790d2fe59174fb1290/controllers/openstackmachine_controller.go#L465
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/6859acb599b3c358e682da790d2fe59174fb1290/pkg/cloud/services/compute/instance.go#L162
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/6859acb599b3c358e682da790d2fe59174fb1290/pkg/cloud/services/compute/instance.go#L176
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/6859acb599b3c358e682da790d2fe59174fb1290/pkg/cloud/services/networking/port.go#L97-L100

So the security groups in `instanceSpec` actually end up being propagated to the ports in the case where the ports don't set their own secgroups, which is what we want. If you think of `instanceSpec` as an overall container for what we want the instance to look like it makes sense. I would probably prefer to leave it as it is TBH.

Reply:
yes, I admin I didn't dig into such detail logic but I checked the code and I agree it's what you proposed way
so let port use sg if given otherwise from instanceSpec..